CMakeLists.txt

```diff
@@ -15,15 +15,19 @@ include_directories(${BOOST_INCLUDE_DIR})
 pkg_check_modules(EIGEN3 REQUIRED "eigen3")
 include_directories(${EIGEN3_INCLUDE_DIRS})
 
-find_package(SISL)
-if (SISL_FOUND)
-    pkg_check_modules(BASELIB REQUIRED "base-lib")
-    include_directories(${BASELIB_INCLUDE_DIRS})
+pkg_check_modules(BASELIBSISL "base-lib-sisl")
+if(BASELIBSISL_FOUND)
+    find_package(SISL REQUIRED)
     include_directories(${SISL_INCLUDE_DIRS})
     list(APPEND TOOLKIT_ADDITIONAL_SOURCES ${CMAKE_SOURCE_DIR}/src/spline.cpp)
-    list(APPEND TOOLKIT_ADDITIONAL_LIBRARIES ${BASELIB_LIBRARIES} ${SISL_LIBRARIES})
+    list(APPEND TOOLKIT_ADDITIONAL_LIBRARIES ${SISL_LIBRARIES})
+    list(APPEND PKG_CFLAGS "-DWITH_SISL=1")
```

Comment:
Rather than this being done implicitly. I would actually prefer adding an explicit `add_definition` to OroGen's typekit definition, and have it code-generate the `add_definitions`. Same for the additional sources. It would place all of the logic in the orogen file, which feels better somehow.

The addition of SISL in TOOLKIT_ADDITIONAL_LIBRARIES feels weird here by the way ... Since we're `using_library` it should not be needed - which would simplify the whole block to only be adding `src/spline.cpp`.

Reply:
> Rather than this being done implicitly. I would actually prefer adding an explicit `add_definition` to OroGen's typekit definition, and have it code-generate the `add_definitions`. Same for the additional sources. It would place all of the logic in the orogen file, which feels better somehow.

You mean the following way:
```
using_library 'base-lib-sisl'
typekit.add_definition("WITH_SISL",1)
```

Reply:
Something like that, yes.